File: LMIS/LMIS.Web/Modules/ServerTypes/BookManage/PublisherColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { PublisherRow } from "./PublisherRow";

export interface PublisherColumns {
    PublisherId: Column<PublisherRow>;
    PublisherName: Column<PublisherRow>;
    Contact: Column<PublisherRow>;
    ContactPhone: Column<PublisherRow>;
    Email: Column<PublisherRow>;
    UpdateTime: Column<PublisherRow>;
}

export class PublisherColumns extends ColumnsBase<PublisherRow> {
    static readonly columnsKey = 'BookManage.Publisher';
    static readonly Fields = fieldsProxy<PublisherColumns>();
}

File: LMIS/LMIS.Web/Modules/BookManage/Publisher/PublisherColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.BookManage.Columns;

[ColumnsScript("BookManage.Publisher")]
[BasedOnRow(typeof(PublisherRow), CheckNames = true)]
public class PublisherColumns
{
    //[EditLink,AlignRight]
    //public long PublisherId { get; set; }
    [EditLink]
    public string PublisherName { get; set; }
    public string Contact { get; set; }
    public string ContactPhone { get; set; }
    [Width(250)]
    public string Email { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}